src/cli/wallet.rs

```diff
@@ -66,36 +67,81 @@ impl WalletOpts {
+                        let wallet = WalletOpts::read_encrypted(&o.opts).await?;
+                        println!("{}", wallet.private(o.index)?);
+                    },
+                }         
             }
             Command::Public(o) => {
-                let wallet = WalletOpts::read(&o.opts).await?;
-                println!("{}", wallet.public(o.index)?);
+                match WalletOpts::read(&o.opts).await {
+                    Ok(wallet) => println!("{}", wallet.public(o.index)?),
+                    _ => {
+                        let wallet = WalletOpts::read_encrypted(&o.opts).await?;
+                        println!("{}", wallet.public(o.index)?);
+                    },
+                }     
             }
             Command::Address(o) => {
-                let wallet = WalletOpts::read(&o.opts).await?;
-                println!("{}", wallet.address(o.index)?);
+                match WalletOpts::read(&o.opts).await {
+                    Ok(wallet) => println!("{}", wallet.address(o.index)?),
+                    Err(_) => {
+                        let wallet = WalletOpts::read_encrypted(&o.opts).await?;
+                        println!("{}", wallet.address(o.index)?);
+                    },
+                }     
+            }
+            Command::Password(o) => {
+                let manager = WalletManager::new(&o.opts.file.as_ref().unwrap());
+                match &o.remove {
+                    true => {
+                        match manager.load_unlocked().await {
+                            Ok(_) => println!("You haven't defined a password"),
+                            Err(_) => manager.decrypt().await?,
+                        }
+                    } 
+                    false => {
+                        match manager.load_unlocked().await {
+                            Ok(_) => manager.encrypt().await?,
+                            Err(_) => manager.reencrypt().await?,
+                        }
+                    }
+                }
             }
         };
         Ok(())
     }
 
     async fn read(o: &CommonOpts) -> anyhow::Result<Wallet> {
-        let manager = WalletManager::new(&o.file);
+        let manager = WalletManager::new(&o.file.as_ref().unwrap());
         let wallet = manager.wallet(&o.wallet_id()?).await?;
         Ok(wallet)
     }
 
+    async fn read_encrypted(o: &CommonOpts) -> anyhow::Result<Wallet> {
+        let password = Password::with_theme(&ColorfulTheme::default())
+            .with_prompt("Enter password")
+            .interact()
+            .unwrap();
+        let manager = WalletManager::new(&o.file.as_ref().unwrap());
+        let wallet = manager.wallet_encrypted(&o.wallet_id()?, &password).await?;
+        Ok(wallet)
+    }
+
     async fn create(o: &CommonOptsCreate) -> anyhow::Result<(WalletManager, WalletId)> {
-        let manager = WalletManager::new(&o.common_opts.file);
+        let mut manager = WalletManager::new("default.wallet");
```

Comment:
Should this be `common_opts.file`?

Reply:
The idea was to create the file "default.wallet" if `None` was provided in `common_opts.file`.

Reply:
Ahh right, all good. Was considering using the user's home directly, but let's do that in another issue.